core/src/main/java/org/stellar/anchor/sep31/Sep31Service.java

```diff
@@ -250,11 +334,31 @@ void validateAndApplyFeeAndQuote(
                 "Quote asset is [%s] different from the transaction sending asset [%s]",
                 quote.getSellAmount(), txn.getAmountIn()));
       }
-      txn.setQuoteId(request.getQuoteId());
     }
   }
 
-  void validateKyc(String senderId, String receiverId) throws AnchorException {
+  void queryFee() throws AnchorException {
+    Sep31PostTransactionRequest request = Context.get().getRequest();
+    JwtToken token = Context.get().getJwtToken();
+    Amount fee =
+        feeIntegration
+            .getFee(
+                GetFeeRequest.builder()
+                    .sendAmount(request.getAmount())
+                    .sendAsset(request.getAssetCode())
+                    .receiveAsset(request.getAssetCode())
```

Comment:
❗ I think this is supposed to be the destination asset.

```suggestion
                    .receiveAsset(request.getDestinationAsset())
```

Reply:
Cool catch. Thanks so much.